src/model.rs

```diff
@@ -221,7 +221,7 @@ impl BrCode {
         encode.push_str(&format!("5303{:02}", self.currency));
         match self.amount {
             None => (),
-            Some(a) => encode.push_str(&format!("54{:02}{}", a.to_string().len(), a)),
+            Some(a) => encode.push_str(&format!("54{:02}{:.2}", a.to_string().len(), a)),
```

Comment:
The length doesn't take into account the additional digits.
```suggestion
            Some(a) => encode.push_str(&format!(
                "54{:02}{:.2}",
                &format!("{:.2}", a).to_string().len(),
                a
            )),```

Reply:
@otaviopace I would save the formatted string before pushing it:

``
Some(a) => {
let a = format!("{:.2}", a);
encode.push_str(&format!(
                "54{:02}{}",
                &a.len(),
                a
            ))},
```

Reply:
97afd35